engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -249,23 +253,36 @@ private void initRenderGraph() {
         // TODO: write snippets and shaders to inspect content of a color/depth buffer
 
         // sky generation
-        FBOConfig reflectedRefractedBufferConfig = new FBOConfig(new ResourceUrn("engine:sceneReflectiveRefractive"),
-                        FULL_SCALE, FBO.Type.HDR).useNormalBuffer();
-        BufferClearingNode.RequiredData reflectiveRefractiveBufferClearingData =
-                new BufferClearingNode.RequiredData(reflectedRefractedBufferConfig, displayResolutionDependentFBOs, GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
-        Node reflectedRefractedClearingNode = nodeFactory.createInstance(BufferClearingNode.class, reflectiveRefractiveBufferClearingData);
+        FBOConfig reflectedRefractedBufferConfig = new FBOConfig(new ResourceUrn("engine:sceneReflectiveRefractive"), FULL_SCALE, FBO.Type.HDR).useNormalBuffer();
+        BufferClearingNode reflectedRefractedClearingNode = nodeFactory.createInstance(BufferClearingNode.class, DELAY_INIT);
+        reflectedRefractedClearingNode.initialise(reflectedRefractedBufferConfig, displayResolutionDependentFBOs, GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
         renderGraph.addNode(reflectedRefractedClearingNode, "reflectedRefractedClearingNode");
 
-        BufferClearingNode.RequiredData readBufferClearingData = new BufferClearingNode.RequiredData(READ_ONLY_GBUFFER.getConfig(), displayResolutionDependentFBOs,
-                        GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
-        Node readBufferClearingNode = nodeFactory.createInstance(BufferClearingNode.class, readBufferClearingData);
+        BufferClearingNode readBufferClearingNode = nodeFactory.createInstance(BufferClearingNode.class, DELAY_INIT);
+        readBufferClearingNode.initialise(READ_ONLY_GBUFFER.getConfig(), displayResolutionDependentFBOs,
+                GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
         renderGraph.addNode(readBufferClearingNode, "readBufferClearingNode");
 
         Node backdropNode = nodeFactory.createInstance(BackdropNode.class);
         renderGraph.addNode(backdropNode, "backdropNode");
 
+        String aLabel = "hazeIntermediateNode";
+        FBOConfig hazeIntermediateConfig = new FBOConfig(HazeNode.INTERMEDIATE_HAZE, ONE_16TH_SCALE, FBO.Type.DEFAULT);
+        HazeNode hazeIntermediateNode = nodeFactory.createInstance(HazeNode.class, DELAY_INIT);
+        hazeIntermediateNode.initialise(READ_ONLY_GBUFFER.getConfig(), hazeIntermediateConfig, aLabel);
+        renderGraph.addNode(hazeIntermediateNode, aLabel);
+
+        aLabel = "hazeFinalNode";
+        FBOConfig hazeFinalConfig = new FBOConfig(HazeNode.FINAL_HAZE, ONE_32TH_SCALE, FBO.Type.DEFAULT);
+        HazeNode hazeFinalNode = nodeFactory.createInstance(HazeNode.class, DELAY_INIT);
+        hazeFinalNode.initialise(hazeIntermediateConfig, hazeFinalConfig, aLabel);
+        renderGraph.addNode(hazeFinalNode, aLabel);
```

Comment:
Eventually I'll probably move entire blocks of node initialization/registration into the graph into their own methods, i.e. `setupShadowNodes()`. Right now it's useful to have everything listed this way for debugging purposes: I can easily comment nodes in/out.